### PR TITLE
Lesson 18

### PR DESCRIPTION
**Pull request by dlincheuski**

### CODE REVIEW

File: lesson 18.playground/Contents.swift [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+
+var singer = Singer(firstName: "Dmitry", surname: "Babushkin")
+
+class Musican: Artist {
+    override func printPerformance() -> String {
+        return super.printPerformance() + "musical perfornance"
+    }
+}
+
+var musican = Musican(firstName: "Vladimir", surname: "Nimerovich")
```
Comment (nograsby) [inline lesson 18.playground/Contents.swift]: i

File: lesson 18.playground/Contents.swift [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+var allPerformances = [dancer, painter, singer, musican]
```
Comment (Nograsby) [inline lesson 18.playground/Contents.swift:69]: perfomances

File: lesson 18.playground/Contents.swift [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+
+for value in allPerformances {
```
Comment (nograsby) [inline lesson 18.playground/Contents.swift]: performance

File: lesson 18.playground/Contents.swift (hunk 2 above)
Comment (nograsby) [inline lesson 18.playground/Contents.swift]: artists

File: lesson 18.playground/Contents.swift [hunk 4]
```diff
@@ -0,0 +1,312 @@
+/* 1. Есть базовый клас артист, у него есть имя и фамилия и есть метод выступление
+создать 4 класса танцор музыкант художник певец которые наследуюся от артиста
+художник при попытке изменить имя должен оставлять свое имя
+при вызове метода выступление должно печататься имя фамилия род занятий(используя полиформизм и массив) */
+
+class Artist {
+    var firstName: String
+    var surname: String
+    
+    func printPerformance() -> String {
```
Comment (nograsby) [inline lesson 18.playground/Contents.swift]: perfomanceInfo

File: lesson 18.playground/Contents.swift [hunk 5]
```diff
@@ -0,0 +1,312 @@
...
+    override func printPerformance() -> String {
+        return super.printPerformance() + "musical perfornance"
+    }
+}
+
+var musican = Musican(firstName: "Vladimir", surname: "Nimerovich")
+var allPerformances = [dancer, painter, singer, musican]
+
+for value in allPerformances {
+    print(value.printPerformance())
```
Comment (nograsby) [inline lesson 18.playground/Contents.swift]: print(artist.performanceInfo)

File: lesson 18.playground/Contents.swift [hunk 6]
```diff
@@ -0,0 +1,312 @@
...
+}
+/* 2. Создать базовый класс транстпортное средство и у него будет 3 свойства скорость вместимовсть и стоимость 1 перевозки
+ создать несколько дочерних классов самолет корабль вертолет машина поезд и у каждого по 1 обьекту
+ определить сколько уйдет денег и времени что бы перевезти 100 и 10000 человек каждым из транспортных средств*/
+class Transport {
+    var speed: Int
+    var capasity: Int
```
Comment (nograsby) [inline lesson 18.playground/Contents.swift]: c

File: lesson 18.playground/Contents.swift [hunk 7]
```diff
@@ -0,0 +1,312 @@
...
+    init(speed: Int, capasity: Int, oneTripCost: Int, numberOfPeople: Int) {
+        self.speed = speed
+        self.capasity = capasity
+        self.oneTripCost = oneTripCost
+        self.numberOfPeople = numberOfPeople
+    }
+    
+}
+
+class Airbus: Transport {
```
Comment (nograsby) [inline lesson 18.playground/Contents.swift]: class Airbus: Transport { }

File: lesson 18.playground/Contents.swift [hunk 8]
```diff
@@ -0,0 +1,312 @@
...
+var train10000 = Train(speed: 75, capasity: 350, oneTripCost: 50, numberOfPeople: 10000)
+
+
+class Car: Transport {
+}
+
+var car100 = Car(speed: 120, capasity: 6, oneTripCost: 75, numberOfPeople: 100)
+var car10000 = Car(speed: 120, capasity: 6, oneTripCost: 75, numberOfPeople: 10000)
+
+var dataBases = [airbus100, helicopter100, train100, car100, airbuss10000, helicopter10000, train10000, car10000]
```
Comment (nograsby) [inline lesson 18.playground/Contents.swift]: transports

File: lesson 18.playground/Contents.swift [hunk 9]
```diff
@@ -0,0 +1,312 @@
...
+var animals = [AnyObject]()
+animals = [human1, human2, monkey1, monkey2, dog1, dog2, giraffe1, giraffe2, crocodile1, crocodile2]
+
+var selectAnimal = [AnyObject]()
+var selectAlive = [AnyObject]()
+var selectLeg = [AnyObject]()
+var selectReptiles = [AnyObject]()
+
+for animal in animals {
+    if let value = animal as? Human {
```
Comment (nograsby) [inline lesson 18.playground/Contents.swift]: confirmedAnimal
alive

File: lesson 18.playground/Contents.swift [hunk 10]
```diff
@@ -0,0 +1,312 @@
...
+    }
+}
+
+var crocodile1 = Crocodile(animal: true, alive: true, leg: 4, reptiles: true)
+var crocodile2 = Crocodile(animal: true, alive: true, leg: 4, reptiles: true)
+
+var animals = [AnyObject]()
+animals = [human1, human2, monkey1, monkey2, dog1, dog2, giraffe1, giraffe2, crocodile1, crocodile2]
+
+var selectAnimal = [AnyObject]()
```
Comment (nograsby) [inline lesson 18.playground/Contents.swift]: selectedAnimals

File: lesson 18.playground/Contents.swift [hunk 11]
```diff
@@ -0,0 +1,312 @@
...
+/* 2. Создать базовый класс транстпортное средство и у него будет 3 свойства скорость вместимовсть и стоимость 1 перевозки
+ создать несколько дочерних классов самолет корабль вертолет машина поезд и у каждого по 1 обьекту
+ определить сколько уйдет денег и времени что бы перевезти 100 и 10000 человек каждым из транспортных средств*/
+class Transport {
+    var speed: Int
+    var capasity: Int
+    var oneTripCost: Int
+    var numberOfPeople: Int
+        
+    var costAllTrip: Int {
```
Comment (nograsby) [inline lesson 18.playground/Contents.swift]: tripCost
totalCost

File: lesson 18.playground/Contents.swift [hunk 12]
```diff
@@ -0,0 +1,312 @@
...
+for value in allPerformances {
+    print(value.printPerformance())
+}
+/* 2. Создать базовый класс транстпортное средство и у него будет 3 свойства скорость вместимовсть и стоимость 1 перевозки
+ создать несколько дочерних классов самолет корабль вертолет машина поезд и у каждого по 1 обьекту
+ определить сколько уйдет денег и времени что бы перевезти 100 и 10000 человек каждым из транспортных средств*/
+class Transport {
+    var speed: Int
+    var capasity: Int
+    var oneTripCost: Int
```
Comment (nograsby) [inline lesson 18.playground/Contents.swift]: ticketCost

File: lesson 18.playground/Contents.swift [hunk 13]
```diff
@@ -0,0 +1,309 @@
...
+    
+    init(firstName: String, surname: String) {
+        self.firstName = firstName
+        self.surname = surname
+    }
+}
+
+class Dancer: Artist {
+    override func performanceInfo() -> String {
+        return super.performanceInfo() + "dance perfornance"
```
Comment (nograsby) [inline lesson 18.playground/Contents.swift]: m

File: lesson 18.playground/Contents.swift [hunk 14]
```diff
@@ -0,0 +1,309 @@
...
+        return super.performanceInfo() + "musical perfornance"
+    }
+}
+
+var musician = Musician(firstName: "Vladimir", surname: "Nimerovich")
+var artists = [dancer, painter, singer, musician]
+
+for artist in artists {
+    print(artist.performanceInfo())
+}
```
Comment (nograsby) [inline lesson 18.playground/Contents.swift:73]: +.

**Review by nograsby — APPROVED** [review]

**Pull request merged**